flightmate/src/main/java/com/flightmate/dao/AirportDao.java

```diff
@@ -0,0 +1,115 @@
+package com.flightmate.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.flightmate.beans.Airport;
+import com.flightmate.dao.DBConnection;
+
+public class AirportDao {
+    
+    private static AirportDao dao;
+
+    private AirportDao() {}
+
+    public static synchronized AirportDao getDao() {
+        if (dao == null) {
+            dao = new AirportDao();
+        }
+        return dao;
+    }
+
+    public List<Airport> getAllAirports() {
+        List<Airport> airports = new ArrayList<>();
+        String sql = "SELECT airport_id, name, location FROM airports";
```

Comment:
Since you're getting all details from the airport, this can be reduced to `SELECT * FROM airports`